table/format.cc

```diff
@@ -403,4 +404,17 @@ Status UncompressBlockContents(const UncompressionInfo& uncompression_info,
                                                    ioptions, allocator);
 }
 
+Status SanitizeDbHostIdProperty(Env* env, std::string& db_host_id) {
+  if (db_host_id == kDbHostId) {
+    db_host_id.clear();
+    db_host_id.resize(kMaxHostNameLen);
```

Comment:
I also find this extremely weird, to the point that you seem to have bugs yourself: using a fixed-length std::string to contain a \0 terminated C-style string. Performance should not be a concern to copy from a stack allocated fixed-size buffer to a std::string.

Reply:
Yes, a temporary buffer is a better idea.